packages/starlight-links-validator/libs/remark.ts

```diff
@@ -81,7 +81,7 @@ export const remarkStarlightLinksValidator: Plugin<[{ base: string; srcDir: URL
             }
           }
 
-          if (node.name !== 'a') {
+          if (node.name !== 'a' && node.name !== 'LinkCard' && node.name !== 'LinkButton') {
```

Comment:
Perhaps it could look for nodes that has the `href` attribute? That way it would scale to custom components as well.

Reply:
This is a good point that I considered during the implementation and decided to postpone as I want to think about it a bit more. If the end-goal is to provide links validation for custom components in MDX, I guess we may need a bit more control, e.g. I've seen `<Link>` component using the `link` prop, etc. Or maybe the goal is just a best effort.

As I was not yet convinced, I ended up removing that and adding it to my todo list instead 😅

Reply:
Another place to consider could be the sidebar section in `astro.config.mjs`

Reply:
Yeah, was just thinking about custom components too. Would probably want users to specify like:
```
customNodes: [
  ['Link', 'link']
  ['IconLink', 'href']
]
```
Or some other syntax supporting setting component name and attribute.